### ISSUE TITLE
Damage Invulnerability Inconsistent with Config Options; Older Versions Crash

### ISSUE THREAD

**Issue by ihymyly** [qxeii/nox-renoxed#32]
Version 2.7.2 crashes for 1.20.5 and 1.20.6

All previous versions, disabling projectile resistance doesn't actually disable it. Arrows "vanish" rather than damaging the target like it should. Please update 1.20.1. I understand you may be busy with other projects but this mod has a lot of genuinely amazing features fit for Vanilla-esque gameplay.

Edit: Entities still keep their immunity to arrow damage_type. Unable to locate source.
![Screenshot 2024-08-18 001918](https://github.com/user-attachments/assets/7a86a8d2-e51d-4167-adff-d999e45e5342)
![Screenshot 2024-08-18 001931](https://github.com/user-attachments/assets/b08f9c9f-0aa0-4ab0-b40b-78aba60f26d3)

**Comment by qxeii** [comment]
I generally don't maintain support for older versions simply because I like always having the newest content (and if i did the workload would increase permanently every time I updated the mod to a new version, as I do not use one of those fancy multi-version libraries). I love this mod as much as you do which is why I took it upon myself to update it, but this is essentially just me making the work I did in order to make a private modpack public.

As you have guessed, I am working on some private non-modding related projects right now and have not actually played modded minecraft in a while. If you would like to make an attempt at backporting/fixing the older versions of this mod I implore you to go ahead! I would say fixing bugs in existing projects is one of the best ways to learn how to code as it's how I started out.

**Comment by ihymyly** [comment]
Yeah, I understand. See, I would, but I possess zero idea where to start (and no coding skills Dx). From my searching, I've discovered the Mixins for the mobs. I hypothesize that straight up deleting the part about the projectile resistances may work, but of course the config is still broken. Some direction would be appreciated.
[
![Screenshot 2024-08-18 003127](https://github.com/user-attachments/assets/11625287-7707-4e30-8998-8ed08a9c683e)
](url)

**Comment by qxeii** [comment]
Are you able to view the git diffs (should have a lot of red and green lines to show removals and additions) of the update that fixes it? That should show you whatever and wherever the change I (or someone else) made was.

(I would also recommend getting IntelliJ IDEA if you don't have it already because it makes parsing through code much easier)

**Comment by ihymyly** [comment]
Yeah that's exactly what I'm doing. Going through the history and hoping I find something that looks like whatever you said lol.

Note: On 1.21, you fixed the issue with the damage immunity, but only for Slimes (at least, the Patchbox Pull Request did). Evokers and Witches still have the same issue, i.e. completely immune to projectiles, and Piercing phases through.

**Comment by ihymyly** [comment]
This issue is persisting in present updates. While the intended feature is that certain entities would still be affected by Piercing enchantments if "Projectile Immunity" is set to true, the mod seems to be granting them invulnerability to all projectile types. 

- In addition, all granted invulnerabilities seem to ignore the config values when set to "false". This means regardless of options, Witches and Evokers cannot be affected by _magic_ and most _critters_ are entirely invulnerable to fall damage.

On the 1.21/2.7.2 (present) version, Slimes are vulnerable to all projectile types again, regardless of config. This means the user is unable to grant them their projectile immunity.
